include/LieGroupControllers/impl/ProportionalController/ControllerBase.h

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file ControllerBase.h
+ * @authors Giulio Romualdi
+ * @copyright This software may be modified and  distributed under the terms of
+ * the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef LIE_GROUP_CONTROLLERS_IMPL_PROPORTIONAL_CONTROLLER_CONTROLLER_BASE_H
+#define LIE_GROUP_CONTROLLERS_IMPL_PROPORTIONAL_CONTROLLER_CONTROLLER_BASE_H
+
+#include <LieGroupControllers/impl/ControllerBase.h>
+#include <tuple>
+
+namespace LieGroupControllers
+{
+template <typename _Derived>
+class ProportionalControllerBase : public ControllerBase<_Derived>
+{
+    using State = typename ControllerBase<_Derived>::State;
+    using Vector = typename ControllerBase<_Derived>::Vector;
+    using Gains = typename ControllerBase<_Derived>::Gains;
+
+    State m_state;
+    State m_desiredState;
+    Vector m_feedForward;
+    Vector m_controlOutput;
+    Gains m_gain;
+
+public:
+    bool setState(const State& state);
+
+    bool setDesiredState(const State& state);
+
+    bool setFeedForward(const Vector& feedForward);
+
+    void computeControlLaw();
+
+    void setGain(const Gains& gain);
+
+    const Vector& getControl() const;
+};
+
+template <typename _Derived> bool ProportionalControllerBase<_Derived>::setState(const State& state)
+{
+    m_state = state;
+    return true;
+}
+
+template <typename _Derived>
+bool ProportionalControllerBase<_Derived>::setDesiredState(const State& state)
+{
+    m_desiredState = state;
+    return true;
+}
+
+template <typename _Derived>
+bool ProportionalControllerBase<_Derived>::setFeedForward(const Vector& feedForward)
+{
+    m_feedForward = feedForward;
+    return true;
+}
+
+template <typename _Derived>
+void ProportionalControllerBase<_Derived>::setGain(const Gains& gain)
+{
+    m_gain = gain;
+}
+
+template <typename _Derived> void ProportionalControllerBase<_Derived>::computeControlLaw()
+{
+    auto error = (m_desiredState.compose(m_state.inverse())).log();
```

Comment:
This is log in vector form as stated in https://github.com/artivis/manif/blob/6d07bc65cc5b25c49f1231021be5e61132e5f777/include/manif/impl/lie_group_base.h#L96.

It would be helpful to add this as documentation since log() here is a sequence of an actual logarithm mapping of the group plus a `vee` operator.

Reply:
Note that this is a left-minus, or `m_desiredState.lminus(m_state);` in `manif` for short.

Reply:
Done in 08b1d8071228d8f6251c2587666597c009eaf645